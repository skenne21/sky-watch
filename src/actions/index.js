export const addCompanyBio = (bio) => ({
  type: 'ADD_COMPANY_BIO',
  bio
})

export const companyError = (message) => {
  type: 'ADD_ERROR_BIO',
  message
}