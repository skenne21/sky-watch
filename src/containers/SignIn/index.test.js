import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './index';
import * as actions from '../../actions';
// import * as mocks from '../../mockData';
/* eslint-disable no-duplicate-imports*/
import  { SignIn } from './index';
/* eslint-enable no-duplicate-imports*/

describe('SignIn', () => {
  let wrapper, user, addUser, history;

  beforeEach(() => {
    history = {push: jest.fn()};
    user = {
      name:'Bob',
      email:'bob@gmail.com',
      id:1
    };
    addUser = jest.fn();

    wrapper = shallow(
      <SignIn
        history={history}
        user={user}
        addUser={addUser}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have a default state of name as a empty string', () => {
    expect(wrapper.state('name')).toEqual('');
  });

  it('Should have a defualt state of email as an empty string', () => {
    expect(wrapper.state('email')).toEqual('');
  });

  it('Should have a defualt state of passowrd as a empty string', () => {
    expect(wrapper.state('error')).toEqual('');
  });

  it('Should call hangleChange and set change to state', () => {
    const event = { target: {
      name: 'email',
      value: 'bob@gmail.com'
    }};

    expect(wrapper.state('email')).toEqual('');
    wrapper.instance().handleChange(event);
    expect(wrapper.state('email')).toEqual('bob@gmail.com');  
  });

  it('Should handleUser and reset state', () => {
    const user = {
      name:'bob',
      email: 'bob@gmail.com', 
      password: 'password'
    };

    wrapper.setState({
      name:'bob',
      email: 'bob@gmail.com', 
      password: 'password'
    });

    expect(wrapper.state('email')).toEqual('bob@gmail.com');
    expect(wrapper.state('password')).toEqual('password');
    expect(wrapper.state('name')).toEqual('bob');

    wrapper.instance().handleUser(user);

    expect(wrapper.state('email')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
    expect(wrapper.state('name')).toEqual('');
  });

  describe('mapStateToProps', () => {
    it('Should map user to state', () => {
      const mockedState = { user: user };
      const expected = { user: user };
      const mapped = mapStateToProps(mockedState);
      expect(mapped).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('Should dispatch addUser with the right params', () => {
      const mockedDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockedDispatch);

      mapped.addUser(user);
      const expected = actions.addUser(user);

      expect(mockedDispatch).toHaveBeenCalledWith(expected);
    });
  });
});