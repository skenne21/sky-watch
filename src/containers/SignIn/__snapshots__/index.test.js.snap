// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    addUser={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    user={
        Object {
            "email": "bob@gmail.com",
            "id": 1,
            "name": "Bob",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
           SIGN INTO YOUR ACCOUNT 
</h2>,
        <input
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="Enter Your Email"
          type="email"
          value=""
/>,
        <input
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="Enter Your Password"
          type="password"
          vaule=""
/>,
        <button>
          Enter Your Account
</button>,
        "",
      ],
      "className": "Signin-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " SIGN INTO YOUR ACCOUNT ",
        },
        "ref": null,
        "rendered": " SIGN INTO YOUR ACCOUNT ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Enter Your Email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Enter Your Password",
          "type": "password",
          "vaule": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter Your Account",
        },
        "ref": null,
        "rendered": "Enter Your Account",
        "type": "button",
      },
      "",
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
             SIGN INTO YOUR ACCOUNT 
</h2>,
          <input
            className="input"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="email"
            value=""
/>,
          <input
            className="input"
            name="password"
            onChange={[Function]}
            placeholder="Enter Your Password"
            type="password"
            vaule=""
/>,
          <button>
            Enter Your Account
</button>,
          "",
        ],
        "className": "Signin-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " SIGN INTO YOUR ACCOUNT ",
          },
          "ref": null,
          "rendered": " SIGN INTO YOUR ACCOUNT ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Your Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Your Password",
            "type": "password",
            "vaule": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter Your Account",
          },
          "ref": null,
          "rendered": "Enter Your Account",
          "type": "button",
        },
        "",
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SignIn Should match the snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    addUser={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    user={
        Object {
            "email": "bob@gmail.com",
            "id": 1,
            "name": "Bob",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
           SIGN INTO YOUR ACCOUNT 
</h2>,
        <input
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="Enter Your Email"
          type="email"
          value=""
/>,
        <input
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="Enter Your Password"
          type="password"
          vaule=""
/>,
        <button>
          Enter Your Account
</button>,
        "",
      ],
      "className": "Signin-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " SIGN INTO YOUR ACCOUNT ",
        },
        "ref": null,
        "rendered": " SIGN INTO YOUR ACCOUNT ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Enter Your Email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Enter Your Password",
          "type": "password",
          "vaule": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter Your Account",
        },
        "ref": null,
        "rendered": "Enter Your Account",
        "type": "button",
      },
      "",
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
             SIGN INTO YOUR ACCOUNT 
</h2>,
          <input
            className="input"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="email"
            value=""
/>,
          <input
            className="input"
            name="password"
            onChange={[Function]}
            placeholder="Enter Your Password"
            type="password"
            vaule=""
/>,
          <button>
            Enter Your Account
</button>,
          "",
        ],
        "className": "Signin-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " SIGN INTO YOUR ACCOUNT ",
          },
          "ref": null,
          "rendered": " SIGN INTO YOUR ACCOUNT ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter Your Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Enter Your Password",
            "type": "password",
            "vaule": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter Your Account",
          },
          "ref": null,
          "rendered": "Enter Your Account",
          "type": "button",
        },
        "",
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
